assets/global.js

```diff
@@ -740,15 +722,6 @@ class SlideshowComponent extends SliderComponent {
     this.addEventListener('mouseleave', this.focusOutHandling.bind(this));
     this.addEventListener('focusin', this.focusInHandling.bind(this));
     this.addEventListener('focusout', this.focusOutHandling.bind(this));
-
-    if (this.querySelector('.slideshow__autoplay')) {
-      this.sliderAutoplayButton = this.querySelector('.slideshow__autoplay');
```

Comment:
Can you explain in more details why the `SlideshowSharedComponent` differs from the `SlideComponent`, for instance? 

As I look at these classes, I'm questioning whether or not we actually need three components instead of two: the `SlideshowSharedComponent` isn't used anywhere - it's just there to be extended. But if it were to be used on its own, it wouldn't work: both the `focusInHandling` and `focusOutHandling` functions are making reference to this `this.sliderAutoplayButton` that no longer exits in this component, as well as `announcementBar`-related elements that are no longer there.

Reply:
I think I can resolve this one as we discussed it during our pairing session.